// osu-api-extended/dist/utility/auth.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.authorize = exports.login = exports.login_lazer = exports.re_login = exports.set_v2 = exports.set_v1 = exports.cache_v2 = exports.cache_v1 = void 0;
const request_1 = require("../utility/request");
const readline_1 = __importDefault(require("readline"));
const open_1 = __importDefault(require("open"));
const credentials = {
    type: 0,
    username: '',
    password: '',
    clientId: 0,
    clientSecret: '',
    redirect_uri: '',
};
exports.cache_v1 = '';
exports.cache_v2 = '';
const set_v1 = (v) => exports.cache_v1 = v;
exports.set_v1 = set_v1;
const set_v2 = (v) => exports.cache_v2 = v;
exports.set_v2 = set_v2;
const isInitial = () => exports.cache_v2 != '';
const save_credentials = (type, obj) => {
    if (type == 1) {
        credentials.type = 1;
        credentials.username = obj.username;
        credentials.password = obj.password;
    }
    ;
    if (type == 2) {
        credentials.type = 2;
        credentials.clientId = obj.clientId;
        credentials.clientSecret = obj.clientSecret;
    }
    ;
    if (type == 3) {
        credentials.type = 3;
        credentials.clientId = obj.clientId;
        credentials.clientSecret = obj.clientSecret;
        credentials.redirect_uri = obj.redirect_uri;
    }
    ;
};
const re_login = async () => {
    if (credentials.type == 1)
        await (0, exports.login_lazer)(credentials.username, credentials.password);
    if (credentials.type == 2)
        await (0, exports.login)(credentials.clientId, credentials.clientSecret);
    if (credentials.type == 3)
        await (0, exports.authorize)(credentials.clientId, credentials.clientSecret, credentials.redirect_uri);
    return true;
};
exports.re_login = re_login;
const login_lazer = async (username, password) => {
    if (!isInitial())
        save_credentials(1, { username, password });
    const { access_token, expires_in } = await (0, request_1.request)('https://osu.ppy.sh/oauth/token', {
        method: 'POST',
        headers: {
            "Accept": "application/json",
            "Content-Type": "application/json",
        },
        data: JSON.stringify({
            username,
            password,
            grant_type: "password",
            client_id: 5,
            client_secret: 'FGc9GAtyHzeQDshWP5Ah7dega8hJACAJpQtw6OXk',
            scope: "*"
        })
    });
    exports.cache_v2 = access_token;
    return { access_token, expires_in };
};
exports.login_lazer = login_lazer;
const login = async (clientId, clientSecret) => {
    if (!isInitial())
        save_credentials(2, { clientId, clientSecret });
    const { access_token, expires_in } = await (0, request_1.request)('https://osu.ppy.sh/oauth/token', {
        method: 'POST',
        headers: {
            "Accept": "application/json",
            "Content-Type": "application/json",
        },
        data: JSON.stringify({
            grant_type: 'client_credentials',
            client_id: clientId,
            client_secret: clientSecret,
            scope: 'public',
            code: 'code',
        })
    });
    exports.cache_v2 = access_token;
    return { access_token, expires_in };
};
exports.login = login;
const authorize = async (clientId, clientSecret, redirect_uri, scope, state) => {
    if (!isInitial())
        save_credentials(3, { clientId, clientSecret, redirect_uri });
    const cl = readline_1.default.createInterface(process.stdin, process.stdout);
    const question = (q) => new Promise((res, rej) => cl.question(q + ': ', (answer) => res(answer)));
    await (0, open_1.default)(`https://osu.ppy.sh/oauth/authorize?client_id=${clientId}&redirect_uri=${redirect_uri}&response_type=code&scope=friends.read%20identify%20public`);
    const code = await question('Paste code here');
    const { access_token, expires_in } = await (0, request_1.request)('https://osu.ppy.sh/oauth/token', {
        method: 'POST',
        headers: {
            "Accept": "application/json",
            "Content-Type": "application/json",
        },
        data: JSON.stringify({
            grant_type: 'authorization_code',
            client_id: clientId,
            client_secret: clientSecret,
            redirect_uri,
            code,
        })
    });
    exports.cache_v2 = access_token;
    return { access_token, expires_in };
};
exports.authorize = authorize;
