// osu-api-extended/dist/api/v2/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.desc = exports.group = exports.notifications = exports.chat = exports.matches = exports.ranking = exports.changelogs = exports.comments = exports.news = exports.backgrounds = exports.site = exports.forum = exports.scores = exports.beatmap = exports.user = void 0;
const activity_1 = __importStar(require("./routes/user/activity"));
const details_1 = __importStar(require("./routes/user/details"));
const details_2 = __importStar(require("./routes/user/me/details"));
const friends_1 = __importStar(require("./routes/user/me/friends"));
const quota_1 = __importStar(require("./routes/user/me/download/quota"));
const list_1 = __importStar(require("./routes/user/list"));
const category_1 = __importStar(require("./routes/user/scores/category"));
const best_1 = __importStar(require("./routes/user/scores/beatmap/best"));
const all_1 = __importStar(require("./routes/user/scores/beatmap/all"));
const category_2 = __importStar(require("./routes/user/beatmaps/category"));
const most_played_1 = __importStar(require("./routes/user/beatmaps/most_played"));
const kudosu_1 = __importStar(require("./routes/user/beatmaps/kudosu"));
exports.user = {
    activity: activity_1.default,
    details: details_1.default,
    me: {
        details: details_2.default,
        friends: friends_1.default,
        download: {
            quota: quota_1.default
        }
    },
    list: list_1.default,
    scores: {
        category: category_1.default,
        beatmap: {
            best: best_1.default,
            all: all_1.default,
        }
    },
    beatmaps: {
        category: category_2.default,
        most_played: most_played_1.default,
        kudosu: kudosu_1.default,
    }
};
const leaderboard_1 = __importStar(require("./routes/beatmap/leaderboard"));
const list_2 = __importStar(require("./routes/beatmap/list"));
const download_1 = __importStar(require("./routes/beatmap/download"));
const set_1 = __importStar(require("./routes/beatmap/set"));
const diff_1 = __importStar(require("./routes/beatmap/diff"));
const search_1 = __importStar(require("./routes/beatmap/search"));
const events_1 = __importStar(require("./routes/beatmap/events"));
const attributes_1 = __importStar(require("./routes/beatmap/attributes"));
const favourites_1 = __importStar(require("./routes/beatmap/favourites"));
const set_2 = __importStar(require("./routes/beatmap/lookup/set"));
const diff_2 = __importStar(require("./routes/beatmap/lookup/diff"));
const details_3 = __importStar(require("./routes/beatmap/discussions/details"));
const posts_1 = __importStar(require("./routes/beatmap/discussions/posts"));
const votes_1 = __importStar(require("./routes/beatmap/discussions/votes"));
exports.beatmap = {
    leaderboard: leaderboard_1.default,
    list: list_2.default,
    download: download_1.default,
    set: set_1.default,
    diff: diff_1.default,
    search: search_1.default,
    events: events_1.default,
    attributes: attributes_1.default,
    favourites: favourites_1.default,
    lookup: {
        set: set_2.default,
        diff: diff_2.default,
    },
    discussions: {
        details: details_3.default,
        posts: posts_1.default,
        votes: votes_1.default,
    },
};
const details_4 = __importStar(require("./routes/scores/details"));
const download_2 = __importStar(require("./routes/scores/download"));
exports.scores = {
    details: details_4.default,
    download: download_2.default,
};
const details_5 = __importStar(require("./routes/forum/topic/details"));
exports.forum = {
    topic: {
        details: details_5.default,
    },
};
const search_2 = __importStar(require("./routes/site/search"));
const wiki_1 = __importStar(require("./routes/site/wiki"));
const list_3 = __importStar(require("./routes/site/spotlights/list"));
exports.site = {
    search: search_2.default,
    wiki: wiki_1.default,
    spotlights: {
        list: list_3.default,
    },
};
const seasonal_1 = __importStar(require("./routes/backgrounds/seasonal"));
exports.backgrounds = {
    seasonal: seasonal_1.default,
};
const list_4 = __importStar(require("./routes/news/list"));
const details_6 = __importStar(require("./routes/news/details"));
exports.news = {
    list: list_4.default,
    details: details_6.default,
};
const list_5 = __importStar(require("./routes/comments/list"));
const details_7 = __importStar(require("./routes/comments/details"));
exports.comments = {
    list: list_5.default,
    details: details_7.default,
};
const list_6 = __importStar(require("./routes/changelogs/list"));
const lookup_1 = __importStar(require("./routes/changelogs/lookup"));
const details_8 = __importStar(require("./routes/changelogs/details"));
exports.changelogs = {
    list: list_6.default,
    lookup: lookup_1.default,
    details: details_8.default,
};
const details_9 = __importStar(require("./routes/ranking/details"));
exports.ranking = {
    details: details_9.default,
};
const list_7 = __importStar(require("./routes/matches/list"));
const details_10 = __importStar(require("./routes/matches/details"));
exports.matches = {
    list: list_7.default,
    details: details_10.default,
};
// import rooms_list, { types as rooms_list_type } from "./routes/rooms/list";
// import rooms_room_details, { types as rooms_room_details_type } from "./routes/rooms/room/details";
// import rooms_room_leaderboard, { types as rooms_room_leaderboard_type } from "./routes/rooms/room/leaderboard";
// import rooms_room_playlist_scores_list, { types as rooms_room_playlist_scores_list_type } from "./routes/rooms/room/playlist/scores/list";
// import rooms_room_playlist_scores_details, { types as rooms_room_playlist_scores_details_type } from "./routes/rooms/room/playlist/scores/details";
// import rooms_room_playlist_scores_user, { types as rooms_room_playlist_scores_user_type } from "./routes/rooms/room/playlist/scores/user";
// export const rooms: {
//   list: rooms_list_type,
//   room: {
//     details: rooms_room_details_type,
//     leaderboard: rooms_room_leaderboard_type,
//     playlist: {
//       scores: {
//         list: rooms_room_playlist_scores_list_type,
//         details: rooms_room_playlist_scores_details_type,
//         user: rooms_room_playlist_scores_user_type,
//       }
//     }
//   }
// } = {
//   list: rooms_list,
//   room: {
//     details: rooms_room_details,
//     leaderboard: rooms_room_leaderboard,
//     playlist: {
//       scores: {
//         list: rooms_room_playlist_scores_list,
//         details: rooms_room_playlist_scores_details,
//         user: rooms_room_playlist_scores_user,
//       }
//     }
//   }
// };
const updates_1 = __importStar(require("./routes/chat/updates"));
const presense_1 = __importStar(require("./routes/chat/presense"));
const list_8 = __importStar(require("./routes/chat/channels/list"));
const details_11 = __importStar(require("./routes/chat/channels/details"));
const list_9 = __importStar(require("./routes/chat/channels/messages/list"));
const send_1 = __importStar(require("./routes/chat/channels/messages/send"));
const readed_1 = __importStar(require("./routes/chat/channels/messages/readed"));
exports.chat = {
    updates: updates_1.default,
    presense: presense_1.default,
    channel: {
        list: list_8.default,
        details: details_11.default,
        messages: {
            readed: readed_1.default,
            list: list_9.default,
            send: send_1.default,
        },
    },
};
const list_10 = __importStar(require("./routes/notifications/list"));
const readed_2 = __importStar(require("./routes/notifications/readed"));
exports.notifications = {
    list: list_10.default,
    readed: readed_2.default,
};
const list_11 = __importStar(require("./routes/group/list"));
exports.group = {
    list: list_11.default,
};
exports.desc = {
    user_activity: activity_1.description,
    user_details: details_1.description,
    user_me_details: details_2.description,
    user_me_friends: friends_1.description,
    user_me_download_quota: quota_1.description,
    user_list: list_1.description,
    user_scores_category: category_1.description,
    user_scores_beatmap_best: best_1.description,
    user_scores_beatmap_all: all_1.description,
    user_beatmap_category: category_2.description,
    user_beatmap_most_played: most_played_1.description,
    user_beatmap_kudosu: kudosu_1.description,
    beatmap_leaderboard: leaderboard_1.description,
    beatmap_list: list_2.description,
    beatmap_download: download_1.description,
    beatmap_set: set_1.description,
    beatmap_diff: diff_1.description,
    beatmap_search: search_1.description,
    beatmap_events: events_1.description,
    beatmap_attributes: attributes_1.description,
    beatmap_favourites: favourites_1.description,
    beatmap_lookup_set: set_2.description,
    beatmap_lookup_diff: diff_2.description,
    beatmap_discussions_details: details_3.description,
    beatmap_discussions_posts: posts_1.description,
    beatmap_discussions_votes: votes_1.description,
    scores_details: details_4.description,
    scores_download: download_2.description,
    forum_topic_details: details_5.description,
    site_search: search_2.description,
    site_wiki: wiki_1.description,
    site_spotlights_list: list_3.description,
    backgrounds_seasonal: seasonal_1.description,
    news_list: list_4.description,
    news_details: details_6.description,
    comments_list: list_5.description,
    comments_details: details_7.description,
    changelogs_list: list_6.description,
    changelogs_lookup: lookup_1.description,
    changelogs_details: details_8.description,
    ranking_details: details_9.description,
    matches_list: list_7.description,
    matches_details: details_10.description,
    chat_updates: updates_1.description,
    chat_presense: presense_1.description,
    chat_channel_list: list_8.description,
    chat_channel_details: details_11.description,
    chat_channel_messages_list: list_9.description,
    chat_channel_messages_send: send_1.description,
    chat_channel_messages_readed: readed_1.description,
    notifications_list: list_10.description,
    notifications_readed: readed_2.description,
    group_list: list_11.description,
};
