// osu-api-extended/dist/api/v2/routes/user/details.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const request_1 = require("../../../../utility/request");
const request = (0, request_1.namespace)('https://osu.ppy.sh/api/v2/');
exports.description = {
    auth: 1,
    title: __filename,
    method: 'GET',
    description: 'Return details about user',
    params: [
        {
            type: 'string/number',
            name: 'user',
            optional: false,
            description: 'id of the user',
        },
        {
            type: 'string/number',
            name: 'mode',
            optional: false,
            description: '\`\`\`osu\`\`\` or \`\`\`fruits\`\`\` or \`\`\`mania\`\`\` or \`\`\`taiko\`\`\`',
        },
        {
            type: 'string/number',
            name: 'key',
            optional: true,
            description: '\`\`\`id\`\`\` or \`\`\`username\`\`\`',
        },
    ],
};
;
const name = async (user, mode, key) => {
    const data = await request(`users/${user}/${mode}`, {
        method: 'GET',
        params: { key },
    });
    return data;
};
exports.default = name;
